src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -0,0 +1,185 @@
+    });
+
+    it('should display the step title in the side pane;', async (() => {
+        const title = nativeElement.querySelector<HTMLHeadingElement>('.marquee-title');
+        expect(title.innerText).toBe('Marquee Wizard');
+    }));
+
+    it('should emit when the button is clicked', async () => {
+
+        const closeSpy = spyOn(fixture.componentInstance, 'close');
+
+        let footer = nativeElement.querySelector('.modal-footer');
+        let buttons = footer.querySelectorAll<HTMLButtonElement>('.button-primary');
+
+        expect(buttons.length).toBe(1);
+
+        // clicking the next button
+        buttons.item(0).click();
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        buttons = footer.querySelectorAll<HTMLButtonElement>('.button-primary');
+
+        expect(buttons.length).toBe(1);
+
+        // clicking the finish button
+        buttons.item(0).click();
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        // the finish event should have emitted
+        expect(closeSpy).toHaveBeenCalled();
+
+      });
+
+      it('should default to the first step', () => {
+        const steps = document.querySelectorAll<HTMLUListElement>('.marquee-wizard-step');
+
+        const firstStep = steps.item(0);
+
+        expect(firstStep).toBeTruthy();
+
+      });
+
+});
+
+/**
+ * Test for EL-3790 where an error can occur if steps rely on request response
+ */
+@Component({
+    selector: 'marquee-wizard-app',
+    template: `
+<ux-marquee-wizard class="marquee-wizard">
+    <ux-marquee-wizard-step *ngFor="let step of steps" [header]="step.title">
+        <h3 class="marquee-step-title m-t-nil">Marquee wizard Step</h3>
+    </ux-marquee-wizard-step>
+</ux-marquee-wizard>`
+})
+export class MarqueeWizardNgForComponent {
+    step: number = 0;
+    steps = [];
+
+    constructor() {
+        setTimeout(() => {
+            this.steps = [
+                { title: 'first step' },
+                { title: 'second step' },
+            ];
+        }, 3000);
+    }
+}
+
+describe('Marquee wizard ngFor example', () => {
+
+    let component: MarqueeWizardComponent;
+    let fixture: ComponentFixture<MarqueeWizardComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [MarqueeWizardModule],
+            declarations: [MarqueeWizardComponent],
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(MarqueeWizardComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should update steps after delay (eg. server delay)', fakeAsync(() => {
+        tick(4000);
+        // see if steps are present now
+
+        const steps = document.querySelectorAll<HTMLUListElement>('.marquee-wizard-step');
+        expect(steps).toBeTruthy();
```

Comment:
Can we assert that there are 2 steps and they have the appropriate content instead of this? We also need to be testing that the `disableNextWhenInvalid` functionality still works with the delayed steps.